packages/rum-core/src/index.js

```diff
@@ -25,25 +25,30 @@
 
 // export public core APIs.
 
-import ErrorLogging from './error-logging'
-import PerformanceMonitoring from './performance-monitoring'
-import ServiceFactory from './common/service-factory'
+import { registerServices as registerErrorServices } from './error-logging'
+import { registerServices as registerPerfServices } from './performance-monitoring'
+import { ServiceFactory } from './common/service-factory'
 import {
   isPlatformSupported,
   scheduleMicroTask,
   scheduleMacroTask
 } from './common/utils'
 import { patchAll, patchEventHandler } from './common/patching'
-import { PAGE_LOAD, ERROR } from './common/constants'
+import {
+  PAGE_LOAD,
+  ERROR,
+  CONFIG_SERVICE,
+  LOGGING_SERVICE,
+  APM_SERVER
+} from './common/constants'
 import { getInstrumentationFlags } from './common/instrument'
 import afterFrame from './common/after-frame'
 import { createTracer } from './opentracing'
 
+registerPerfServices()
```

Comment:
this feels like Side Effect. can we move this inside createServiceFactory itself? Atleast we know we register only when someone creates the factory.